G/GDAL/build_tarballs.jl

```diff
@@ -132,20 +135,20 @@ hdf5_platforms = expand_cxxstring_abis(hdf5_platforms)
 
 # Dependencies that must be installed before this package can be built
 dependencies = [
-    Dependency("GEOS_jll"; compat="~3.11"),
-    Dependency("PROJ_jll"; compat="~900.100"),
+    Dependency("GEOS_jll"; compat="3.11.2"),
+    Dependency("PROJ_jll"; compat="900.100.0"),
```

Comment:
If #7366 and #7367 are merged, it would be good to switch to:

```suggestion
    Dependency("PROJ_jll"; compat="901"),
```